Display related orders table when viewing the new "Edit Order" page (HPOS enabled stores)

Jinksi opened the pull request:
Fixes #192 

## Description

<!--
Write a brief summary about this PR. 
- Why is this change needed? 
- What does this change do? 
- Were there other solutions you considered? 
- Why did you choose to pursue this solution? 
- Describe any trade-offs you might have had to make.
-->

This PR displays the related orders table meta box on the "Edit Order" and "Edit Subscription" when HPOS is enabled with non-HPOS compatibility.

In addition:
- [x] PHPCS formatting has been fixed in related files.
	(mostly moved to #280)
- [x] PHPCS errors/warnings have been addressed in related files.
	(mostly moved to #280)
	- Some low-hanging errors/warnings have been addressed.
- [x] Added deprecated functions, hooks or classes to the [spreadsheet](https://docs.google.com/spreadsheets/d/1xw9xszcPMnWsp4C8OKZMsLzZob7tOmWT7qMqmEIq314/edit#gid=0) - yes, 3 hooks added

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos as appropriate.
-->

## How to test this PR

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
Use the testing instructions from the linked issue as a starting point.
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Enable the COT feature and turn on COT ([see guide here](https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#easy-way)).
2. Go into **WooCommerce > Settings > Advanced > Custom Data Stores**.
3. Make sure "Use the WooCommerce orders tables" is enabled.
4. Purchase a subscription and note the order ID.
5. From WP admin dashboard, go to **WooCommerce > Orders** and view the "Edit Order" page for the order ID noted above.
6. Notice that the Related Orders table meta box is visible and contains a link to the related Subscription.
   You should not see the current order within this table.
	<img width="792" alt="Screenshot 2022-11-08 at 12 57 37" src="https://user-images.githubusercontent.com/3147296/200465243-366995cc-6fe1-4345-b38b-90238d065660.png">	
7. Click on the related Subscription to be taken to the "Edit Subscription" page.
   You should not see the current subscription within this table.
	<img width="795" alt="Screenshot 2022-11-08 at 12 57 24" src="https://user-images.githubusercontent.com/3147296/200465254-3bc410fd-046e-4035-8557-988f917f8f56.png">
9. Notice that the Related Orders table meta box is visible and contains a link to the parent order.
10. Clicking on the parent order will take you to the "Edit Order" page as above.


Recommended compatibility tests (checked = tested by PR author):
- [x] WCPay
- [x] WC Subs
	- [x] with WC Subs activated + HPOS enabled + syncing on, ~the Related Orders table isn't rendered correctly and following PHP Notice occurs~ it uses a newly deprecated hook. Need to update this issue to clarify:
	https://github.com/woocommerce/woocommerce-subscriptions/issues/4417
- [x] WC 6.8.0 – toggling between this subs-core branch & `trunk` results in identical output.
- [x] WC HPOS enabled + syncing off
- [x] WC HPOS enabled + syncing on
- [x] WC HPOS disabled

## Product impact
<!-- What products will this PR ship in? -->

- [x] Added changelog entry (or does not apply)
- [ ] Will this PR affect WooCommerce Subscriptions? yes/no/tbc, add issue ref
- [ ] Will this PR affect WooCommerce Payments? yes/no/tbc, add issue ref

Jinksi committed:
Enqueue admin styles on HPOS edit order page

Jinksi committed:
Provide HPOS shop-order screen id to add_meta_box

Jinksi committed:
Refactor related metabox class to support HPOS orders

mattallan commented on includes/admin/class-wcs-admin-meta-boxes.php:
```diff
@@ -85,8 +87,12 @@ public function add_meta_boxes() {
-			add_meta_box( 'subscription_renewal_orders', __( 'Related Orders', 'woocommerce-subscriptions' ), 'WCS_Meta_Box_Related_Orders::output', 'shop_order', 'normal', 'low' );
+		if ( wcs_order_contains_subscription( $post_or_order_object, 'any' ) ) {
+			// If HPOS is enabled, we need to get the screen id and provide it to add_meta_box().
+			$screen = wc_get_container()->get( CustomOrdersTableController::class )->custom_orders_table_usage_is_enabled()
```
We have a helper function for this now: `wcs_is_custom_order_tables_usage_enabled()`

mattallan replied:
This will resolve any issues on older versions of WC where `CustomOrdersTableController` doesn't exist.

Jinksi replied:
Nice! I knew that line would break spectacularly when I tested backwards-compat but didn't want to wrap it in yet another if/ternary 😂

Jinksi replied:
Resolved in 195a29e

Jinksi committed:
Utilise COT enabled helper function

Jinksi committed:
Remove unused import

Jinksi committed:
Merge branch 'trunk' into fix/192-hpos-related-orders-metabox

Jinksi committed:
Add changelog entry

Jinksi committed:
Fix PHPCS formatting to improve code quality

Jinksi committed:
Merge branch 'trunk' into fix/192-hpos-related-orders-metabox

Jinksi committed:
Fix type in comment

Jinksi committed:
Remove author from docblocks to fix PHPCS errors

Jinksi committed:
Provide version and in_footer args to admin script

to fix PHPCS errors/warnings and improve browser caching.

$ver set to subs-core version string.

$in_footer set to `false` to match current enqueue behaviour.

Jinksi commented on includes/admin/class-wcs-admin-meta-boxes.php:
```diff
@@ -112,36 +113,44 @@ public function remove_meta_box_save( $post_id, $post ) {
 	 */
 	public function enqueue_styles_scripts() {
 		global $post;
+		$ver = WC_Subscriptions_Core_Plugin::instance()->get_plugin_version();
```
This allows browser-cached scripts to be refreshed when this version number is updated.

Jinksi commented on includes/admin/class-wcs-admin-meta-boxes.php:
```diff
@@ -112,36 +113,44 @@ public function remove_meta_box_save( $post_id, $post ) {
 		if ( 'shop_subscription' == $screen_id ) {
 
-			wp_register_script( 'jstz', WC_Subscriptions_Core_Plugin::instance()->get_subscriptions_core_directory_url( 'assets/js/admin/jstz.min.js' ) );
+			wp_register_script( 'jstz', WC_Subscriptions_Core_Plugin::instance()->get_subscriptions_core_directory_url( 'assets/js/admin/jstz.min.js' ), [], $ver, false );
```
Passing `$in_footer` as `false` to match current (default) enqueue behaviour.

Jinksi commented on includes/admin/meta-boxes/views/html-related-orders-row.php:
```diff
@@ -11,52 +11,52 @@
 <tr>
 	<td>
-		<a href="<?php echo esc_url( get_edit_post_link( $order->get_id() ) ); ?>">
+		<a href="<?php echo esc_url( get_edit_post_link( $row_order->get_id() ) ); ?>">
```
Changed to `$row_order` to remove confusion between the global/current order and related orders in each row.

james-allan commented on includes/admin/class-wc-subscriptions-admin.php:
```diff
@@ -1203,29 +1214,32 @@ public static function has_settings() {
 
-		return apply_filters( 'woocommerce_subscription_settings', array(
-
+		return apply_filters(
```
```suggestion
		/**
		 * Filter the settings for the Subscriptions extension.
		 *
		 * @param array $settings Array of settings in the format required by the woocommerce_admin_fields() function.
		 */
		return apply_filters( 'woocommerce_subscription_settings', array(
```

Adds a filter comment block.

james-allan commented on includes/admin/meta-boxes/class-wcs-meta-box-related-orders.php:
```diff
@@ -21,47 +20,47 @@ class WCS_Meta_Box_Related_Orders {
+		include_once dirname( __FILE__ ) . '/views/html-related-orders-table.php';
 
-		do_action( 'woocommerce_subscriptions_related_orders_meta_box', $order, $post );
+		do_action( 'woocommerce_subscriptions_related_orders_meta_box', $order );
```
We won't be able to remove the second here as third-parties hooking onto it may be using it and it would cause fatals. Unfortunately, we will need to deprecate the whole filter and continue passing a `post` object as long as we can (while we're still using CPT).

Jinksi replied:
Deprecated in 25c31be.

james-allan commented on includes/admin/meta-boxes/class-wcs-meta-box-related-orders.php:
```diff
@@ -21,47 +20,47 @@ class WCS_Meta_Box_Related_Orders {
-			$this_subscription = wcs_get_subscription( $post->ID );
+		if ( $is_subscription ) {
+			$subscription      = wcs_get_subscription( $order );
+			$this_subscription = wcs_get_subscription( $order->get_id() );
```
These two lines do the same thing right?

Jinksi replied:
Thanks, resolved in cd5e649

james-allan reviewed (changes requested):
I've done a pass over the changes for Code review. I left a couple of comments. Next I'll check this branch out and begin testing.

james-allan commented on includes/admin/meta-boxes/class-wcs-meta-box-related-orders.php:
```diff
@@ -115,24 +114,24 @@ public static function output_rows( $post ) {
 		}
 
-		$orders_to_display = apply_filters( 'woocommerce_subscriptions_admin_related_orders_to_display', $orders_to_display, $subscriptions, $post );
+		$orders_to_display = apply_filters( 'woocommerce_subscriptions_admin_related_orders_to_display', $orders_to_display, $subscriptions, $order );
```
This change is effectively the reason behind the error you mentioned in the PR description and in the corresponding issue (https://github.com/woocommerce/woocommerce-subscriptions/issues/4417):

```
PHP Notice:  Function ID was called <strong>incorrectly</strong>. Order properties should not be accessed directly. Backtrace: require('wp-admin/edit-form-advanced.php'), do_meta_boxes, WCS_Meta_Box_Related_Orders::output, include_once('/plugins/woocommerce-subscriptions-core/includes/admin/meta-boxes/views/html-related-orders-table.php'), do_action('woocommerce_subscriptions_related_orders_meta_box_rows'), WP_Hook->do_action, WP_Hook->apply_filters, WCS_Meta_Box_Related_Orders::output_rows, apply_filters('woocommerce_subscriptions_admin_related_orders_to_display'), WP_Hook->apply_filters, WC_Subscriptions_Switcher::display_switches_in_related_order_metabox, WC_Abstract_Legacy_Order->__get, wc_doing_it_wrong Please see <a href="https://wordpress.org/support/article/debugging-in-wordpress/">Debugging in WordPress</a> for more information. (This message was added in version 3.0.) in //wp-includes/functions.php on line 5835
```

This is a breaking change and rather than replace the `$post` variable with an order object we will need to deprecate this filter, only call it if there are third-parties hooking onto it and then trigger a new hook where we can pass the order freely. 

This will look like this roughly. 

```php
if ( has_filter( 'woocommerce_subscriptions_admin_related_orders_to_display' ) ) {
	wcs_deprecated_hook( 'woocommerce_subscriptions_admin_related_orders_to_display', 'subscriptions-core 5.0.0', 'wcs_admin_subscription_related_orders_to_display' );

	/**
	 * Filters the orders to display in the Related Orders meta box.
	 *
	 * This filter is deprecated in favour of 'wcs_admin_subscription_related_orders_to_display'.
	 *
	 * @deprecated subscriptions-core 5.0.0
	 *
	 * @param array   $orders_to_display An array of orders to display in the Related Orders meta box.
	 * @param array   $subscriptions An array of subscriptions related to the order.
	 * @param WP_Post $post The order post object.
	 */
	$orders_to_display = apply_filters( 'woocommerce_subscriptions_admin_related_orders_to_display', $orders_to_display, $subscriptions, get_post( $order->get_id() ) );
}

/**
 * Filters the orders to display in the Related Orders meta box.
 *
 * @since subscriptions-core 5.0.0
 *
 * @param array    $orders_to_display An array of orders to display in the Related Orders meta box.
 * @param array    $subscriptions An array of subscriptions related to the order.
 * @param WC_Order $order The order object.
 */
$orders_to_display = apply_filters( 'wcs_admin_subscription_related_orders_to_display', $orders_to_display, $subscriptions, $order );
```

james-allan replied:
The `get_post( $order->get_id() )`  will return the order post or the placeholder post on HPOS sites, so on HPOS sites there may be some change in data received, however fully fledged subscription stores running HPOS is far enough down the track that third-parties will likely have replaced uses of this.

This behaviour of post objects becoming unusable is inevitable.

Jinksi replied:
Thanks for the guidance here @james-allan, appreciate it!

I've resolved this one using your suggestion above in 76eb615, noting that this may break for HPOS-enabled sites using the deprecated hook.

james-allan commented on includes/admin/meta-boxes/views/html-related-orders-table.php:
```diff
@@ -22,7 +22,7 @@
 		</thead>
 		<tbody>
-			<?php do_action( 'woocommerce_subscriptions_related_orders_meta_box_rows', $post ); ?>
+			<?php do_action( 'woocommerce_subscriptions_related_orders_meta_box_rows', $order ); ?>
```
assuming that this use to pass a post object, we'll need to deprecate this one too unfortunately.

Jinksi replied:
Deprecated in d5da9de.

Jinksi commented:
> I've done a pass over the changes for Code review. I left a couple of comments. Next I'll check this branch out and begin testing.

Thanks for the feedback ☺️ I'll test it all again after the code changes you've suggested. You may as well delay your testing until then as well 🤷‍♂️

Jinksi committed:
Merge branch 'trunk' into fix/192-hpos-related-orders-metabox

Jinksi committed:
Add filter comment block

Co-authored-by: James Allan <james.allan@automattic.com>

Jinksi committed:
Fix formatting

Jinksi committed:
Remove redundant variable

Jinksi commented on includes/admin/class-wcs-admin-meta-boxes.php:
```diff
@@ -112,36 +113,44 @@ public function remove_meta_box_save( $post_id, $post ) {
 	 */
 	public function enqueue_styles_scripts() {
 		global $post;
+		$ver = WC_Subscriptions_Core_Plugin::instance()->get_library_version();
```
Updated to new getter method introduced in #254

Jinksi committed:
Fix formatting

Jinksi committed:
Merge branch 'trunk' into fix/192-hpos-related-orders-metabox

Jinksi committed:
Deprecate hook: related orders to display filter

to allow HPOS support while maintaining backwards compatibility.

Jinksi committed:
Update filter deprecated version tag

Jinksi committed:
Deprecate action: related orders meta box rows

to allow HPOS support while maintaining backwards compatibility.

Jinksi committed:
Deprecate action: related orders meta box

to allow HPOS support while maintaining backwards compatibility.

Jinksi committed:
Change changelog entry to Update to clarify impact

This PR doesn't add anything new, updating to support HPOS.

Jinksi commented:
@james-allan I've marked this as ready for another code review, since I've addressed your comments and deprecated hooks. I will continue testing once we have a resolution for #265, since this branch is affected by it.

Jinksi commented on includes/admin/meta-boxes/views/html-related-orders-row.php:
```diff
@@ -11,52 +11,52 @@
 
 		// Backwards compatibility for third-parties using the generic WP post time filter.
-		$date_to_display = apply_filters( 'post_date_column_time', $date_to_display, get_post( $order->get_id() ) );
+		$date_to_display = apply_filters( 'post_date_column_time', $date_to_display, get_post( $row_order->get_id() ) );
```
Let's deprecate this filter, as it accepts a post parameter. What do you think?

james-allan replied:
Yep that sounds right to me.

Jinksi replied:
Actually, it'd be good to get your advice here. We don't own this filter (it's a WP filter). 

We could wrap this in an if statement, only fire this if HPOS disabled?

james-allan replied:
ah in that case we don't have many options. 

Yeah wrap it in a HPOS check or we have to just stop using it entirely.

Jinksi replied:
👍 updated in 92fac53:

```php
		if ( ! wcs_is_custom_order_tables_usage_enabled() ) {
			// Backwards compatibility for third-parties using the generic WP post time filter.
			// Only apply this filter if HPOS is not enabled, as the filter is not compatible with HPOS.
			$date_to_display = apply_filters( 'post_date_column_time', $date_to_display, get_post( $row_order->get_id() ) );
		}
```

haszari commented:
> I've marked this as ready for another code review, since I've addressed your comments and deprecated hooks. I will continue testing once we have a resolution for https://github.com/Automattic/woocommerce-subscriptions-core/issues/265, since this branch is affected by it.

FYI @Jinksi I've added `Blocked` label, assuming that this PR is blocked by the infinite recursion issue:

- https://github.com/Automattic/woocommerce-subscriptions-core/issues/265

Jinksi committed:
Merge branch 'trunk' into fix/192-hpos-related-orders-metabox

Jinksi commented:
> FYI @Jinksi I've added `Blocked` label, assuming that this PR is blocked by the infinite recursion issue:

Thanks @haszari. This issue has now been fixed in #272, so I'll remove the `Blocked` label and begin another round of testing.

Jinksi commented:
@james-allan It looks like there is now an issue with the metabox not displaying on the Edit Orders screen with HPOS enabled. I'll let you know once it has been resolved.


[`wcs_order_contains_subscription`](https://github.com/Automattic/woocommerce-subscriptions-core/blob/d334429720a4237e4e917c51d55bc31125aa11cc/includes/wcs-order-functions.php#L327) is now returning `false` on the Edit Order screen 🤔

I've added a `Blocked` label due to this PR. A fix has been included in https://github.com/Automattic/woocommerce-subscriptions-core/pull/258#discussion_r1019899441.

Jinksi committed:
Merge branch 'trunk' into fix/192-hpos-related-orders-metabox

Jinksi commented:
I've removed the `Blocked` label since #258 has been merged into `trunk`.

This PR is ready for review! (again)

Jinksi committed:
Use strict equals to fix phpcs warning

Jinksi committed:
Use $this_order to avoid mutated $order variable when passing to filters.

$order is mutated throughout this function, hard to follow it's value.

Adding $this_order is clearer when used further down the function.

Jinksi committed:
Fix related orders metabox rendering on unrelated screens

Jinksi commented on includes/admin/class-wcs-admin-meta-boxes.php:
```diff
@@ -85,8 +84,10 @@ public function add_meta_boxes() {
-			add_meta_box( 'subscription_renewal_orders', __( 'Related Orders', 'woocommerce-subscriptions' ), 'WCS_Meta_Box_Related_Orders::output', 'shop_order', 'normal', 'low' );
+		if ( wcs_order_contains_subscription( $post_or_order_object, 'any' ) ) {
+			// If HPOS is enabled, we need to get the screen id and provide it to add_meta_box().
+			$screen = wcs_is_custom_order_tables_usage_enabled() ? wc_get_page_screen_id( 'shop-order' ) : 'shop_order';
```
Added this `$current_screen->id === $screen` condition, as it was rendering on other screens (e.g. Edit Product).

Jinksi commented on includes/admin/class-wcs-admin-meta-boxes.php:
```diff
@@ -84,9 +83,13 @@ public function add_meta_boxes() {
+		// Get "Edit Order" screen ID, which differs if HPOS is enabled.
+		$screen         = wcs_is_custom_order_tables_usage_enabled() ? wc_get_page_screen_id( 'shop-order' ) : 'shop_order';
+		$current_screen = get_current_screen();
+
```
Added this `$current_screen->id === $screen` condition, as it was rendering on other screens (e.g. Edit Product).

Jinksi commented on includes/admin/class-wcs-admin-meta-boxes.php:
```diff
@@ -84,9 +83,13 @@ public function add_meta_boxes() {
+		$screen         = wcs_is_custom_order_tables_usage_enabled() ? wc_get_page_screen_id( 'shop-order' ) : 'shop_order';
+		$current_screen = get_current_screen();
+
+		// Only display the meta box if viewing an order that contains a subscription.
+		if ( $current_screen->id === $screen && wcs_order_contains_subscription( $post_or_order_object, 'any' ) ) {
+			add_meta_box( 'subscription_renewal_orders', __( 'Related Orders', 'woocommerce-subscriptions' ), 'WCS_Meta_Box_Related_Orders::output', $screen, 'normal', 'low' );
```
Added this `$current_screen->id === $screen` condition, as it was rendering on other screens (e.g. Edit Product).

james-allan replied:
This condition use to be `'shop_order' === WC_Data_Store::load( 'order' )->get_order_type( $post_ID )` was that returning true on product pages?

Jinksi replied:
Before 59f50649ad81a3bca51a8740350d6529a46f79c2, I had the statement:

`if ( wcs_order_contains_subscription( $post_or_order_object, 'any' ) ) {`

This was returning throwing an error on product pages since I wasn't checking if it was the edit order screen (rookie WP hook user mistake).

`'shop_order' === WC_Data_Store::load( 'order' )->get_order_type( $post_ID )` was removed, since it was accessing `$post_ID` and I removed all references to `$post` or `$post_ID`.

Jinksi replied:
I need to double-check `wc_get_page_screen_id` will work with the earliest supported WC version.

james-allan commented on includes/admin/class-wcs-admin-meta-boxes.php:
```diff
@@ -72,10 +71,10 @@ public function __construct() {
-	public function add_meta_boxes() {
-		global $post_ID;
-
+	public function add_meta_boxes( $post_type, $post_or_order_object ) {
```
While it is unlikely, we can't add new params to publically accessible functions like this in case a third party is calling it.

To introduce new params to this function in a backwards compatible way we need to add placeholders. 

eg

```php
public function add_meta_boxes( $post_type = '', $post_or_order_object = null ) {
```

Then inside the function we'll need to get the object from the global post ID or something if this function receives no object.

Jinksi replied:
Ah thanks, I didn't consider backwards compat for `public` functions.

Addressed in 23a1429.

james-allan commented on includes/admin/meta-boxes/views/html-related-orders-row.php:
```diff
@@ -2,7 +2,7 @@
  * Display a row in the related orders table for a subscription or order
  *
- * @var array $order A WC_Order or WC_Subscription order object to display
+ * @var WC_Order|WC_Subscription $row_order A WC_Order or WC_Subscription order object to display.
```
Unfortunately this isn't something we can change so easily without impacting third-parties backwards compat. 

We can either return to using the `$order` arg or if we want to rename this args we would need to add to this file something like:


```
// Comment saying this is for backwards compat reasons. 
if ( ! isset( $row_order ) ) {
    $row_order = $order
}

```

Jinksi replied:
I changed this mostly to address my readability concerns, i.e. is this the "global" order for the page or the order for a single row?

But since this will impact backwards compat, I'll revert this to `$order`. Thanks 🙌

Jinksi replied:
Resolved in d423542.

Do you think it is safe to keep the `WC_Order|WC_Subscription` type change in the docblock?

james-allan commented on includes/admin/meta-boxes/views/html-related-orders-table.php:
```diff
@@ -22,7 +23,31 @@
+				 *
+				 * @param WC_Post $post The order post object.
+				 */
+				do_action( 'woocommerce_subscriptions_related_orders_meta_box_rows', wc_get_order( $post ) );
```
is this suppose to just be `$post`? 
```suggestion
				do_action( 'woocommerce_subscriptions_related_orders_meta_box_rows', $post );
```

Jinksi replied:
Yep! Thanks, resolved in 1881951

james-allan reviewed (changes requested):
Hey Eric, this PR tests well. I've just left a couple of comments regarding some potential backwards compatibility issues.

Jinksi committed:
Merge branch 'trunk' into fix/192-hpos-related-orders-metabox

Jinksi committed:
Include reference to add_meta_boxes WP hook in docblock

Jinksi committed:
Ensure add_meta_boxes() works if no order object provided

for backwards compatibility.

Jinksi committed:
Fix inadvertent change to deprecated hook $post param

Jinksi committed:
Only apply post_date_column_time filter if HPOS disabled

Jinksi committed:
Merge branch 'trunk' into fix/192-hpos-related-orders-metabox

Jinksi committed:
Rename $row_order to $order to maintain backwards compat

Jinksi committed:
Merge branch 'trunk' into fix/192-hpos-related-orders-metabox

Jinksi committed:
Merge branch 'trunk' into fix/192-hpos-related-orders-metabox

james-allan reviewed (approved):
Changes look good to me. Thanks for your patience working on this one @Jinksi 💯 

I tested: 

**HPOS:**
- Viewing the edit order screen of a **related order** shows the related orders table. 
- Viewing the edit order screen of a non-related order doesn't show the screen.

**WP Posts:**
- Viewing the edit order screen of a **related order** shows the related orders table. 
- Viewing the edit order screen of a non-related order doesn't show the screen.

Deprecated notices:

- With the WC Subscriptions plugin enabled, I can see the follow deprecated notice:

```
PHP Deprecated:  Hook woocommerce_subscriptions_admin_related_orders_to_display is <strong>deprecated</strong> since version subscriptions-core 5.1.0! Use wcs_admin_subscription_related_orders_to_display instead. in /Users/jamesallan/local-dev/hpos/app/public/wp-includes/functions.php on line 5758
PHP Stack trace: ...
```

So the deprecated warning are working. 👍 There's an issue for this already over on WC Subscriptions. https://github.com/woocommerce/woocommerce-subscriptions/issues/4417

Note: The links in this table are incorrect, however I understand that will be fixed with https://github.com/Automattic/woocommerce-subscriptions-core/pull/269

Jinksi committed:
Merge branch 'trunk' into fix/192-hpos-related-orders-metabox
